receivers/slack/slack.go

```diff
@@ -315,6 +315,10 @@ func (sn *Notifier) createSlackMessage(ctx context.Context, alerts []*types.Aler
 		}
 		sn.log.Warn("Truncated title", "key", key, "max_runes", slackMaxTitleLenRunes)
 	}
+	if tmplErr != nil {
+		sn.log.Warn("failed to template Slack title", "error", tmplErr.Error())
+		tmplErr = nil
```

Comment:
The only other place I can see where `tmplErr` is referenced is [here](https://github.com/grafana/alerting/pull/248/files#diff-6a83ff1168035e0a58c838a0b2a87e9cd9950ad60e16d801c2d20e45cfff1714R356-R358), which just does a warn log also. I think we could remove that and skip setting it to `nil` here. WDYT?

Reply:
That wouldn't work. If we have an error here and skip setting it to nil, the fields we try to template after this one will be blank, which is causing the `invalid_arguments` errors.
The important thing about this PR is that we're clearing the error before templating the `channel` field.